pyFAI/ext/CSC_common.pxi

```diff
@@ -0,0 +1,308 @@
+                                                 variance=cvariance[idx] if do_variance else 0.0,
+                                                 dark=cdark[idx] if do_dark else 0.0,
+                                                 flat=cflat[idx] if do_flat else 1.0,
+                                                 solidangle=csolidangle[idx] if do_solidangle else 1.0,
+                                                 polarization=cpolarization[idx] if do_polarization else 1.0,
+                                                 absorption=cabsorption[idx] if do_absorption else 1.0,
+                                                 mask=0, #masked pixel have already been treated
+                                                 dummy=cdummy,
+                                                 delta_dummy=cddummy,
+                                                 check_dummy=check_dummy,
+                                                 normalization_factor=normalization_factor,
+                                                 dark_variance=0.0)
+                if not is_valid:
+                    continue
+                for j in range(start, stop):
+                    bin = self._indices[j]
+                    coef = self._data[j]
+                    w = coef * value.norm
+                    if do_azimuthal_variance:
+                        if sum_norm_sq[bin] <= 0.0:
+                            sum_sig[bin] = coef * value.signal
+                            sum_norm[bin] = w
+                            sum_norm_sq[bin] = w * w
+                            sum_count[bin] = coef * value.count
+                        else:
+                            # Inspired from https://dbs.ifi.uni-heidelberg.de/files/Team/eschubert/publications/SSDBM18-covariance-authorcopy.pdf
+                            # Not correct, Inspired by VV_{A+b} = VV_A + ω²·(b-V_A/Ω_A)·(b-V_{A+b}/Ω_{A+b})
+                            # Emprically validated against 2-pass implementation in Python/scipy-sparse   
+                            
+                            omega_A = sum_norm[bin]
+                            omega_B = w
+                            omega2_A = sum_norm_sq[bin]
+                            omega2_B = omega_B*omega_B
+                            sum_norm[bin] = omega_A + omega_B
+                            sum_norm_sq[bin] = omega2_A + omega2_B
+                            # omega3 = acc_norm * omega_A * omega2_B
+                            # VV_{AUb} = VV_A + ω_b^2 * (b-<A>) * (b-<AUb>)
+                            b = value.signal / value.norm
+                            delta1 = sum_sig[bin]/omega_A - b
+                            sum_sig[bin] += coef * value.signal 
+                            delta2 = sum_sig[bin] / sum_norm[bin] - b                        
+                            sum_var[bin] += omega2_B * delta1 * delta2
+                            sum_count[bin] += coef * value.count
+                    else:
```

Comment:
Are there 5 integrations being done here? Above it says "error_model=ErrorModel.NO", so I would expect some of them could be skipped?

Also, I had in mind that only `sum_sig` changes when processing a series of frames with a constant mask. When doing dynamic masking, there should be an alternative route that keeps cached sums and subtracts for the w/count of masked pixels.

Reply:
If ErrorModel is "no" one can save the histogram of variance for sure and one could save the squared norm as well since it is only used to calculate the std but I also use it internally to check the emptiness of bins. I already encounter a case where the normalization factor (extracted from a beam-stop diode) was negative and crashing ODA pocessing when tested without beam. Since it is one write without (additional) read I prefer to keep it.

So yes, one could save one read and one write.